File: 自定义非交互式转场动画/自定义非交互式转场动画/CustomAnimator.m
//
//  CustomAnimator.m
//  自定义非交互式转场动画
//
//  Created by BoBo on 17/2/22.
//  Copyright © 2017年 Li Sen. All rights reserved.
//

#import "CustomAnimator.h"

@implementation CustomAnimator 

//设置转场动画时间
- (NSTimeInterval)transitionDuration:(id<UIViewControllerContextTransitioning>)transitionContext {
    return 2.0;
}
//设置转场动画
- (void)animateTransition:(id<UIViewControllerContextTransitioning>)transitionContext {
    UIView *containerView = [transitionContext containerView];
    
    UIView *toView = [transitionContext viewForKey:UITransitionContextToViewKey];
    UIView *fromView = [transitionContext viewForKey:UITransitionContextFromViewKey];
    
    [containerView addSubview:fromView];
    [containerView addSubview:toView];
    toView.alpha = 0.0;
    
    [UIView animateWithDuration:2.0 animations:^{
        toView.alpha = 1.0;
    } completion:^(BOOL finished) {
        [transitionContext completeTransition:true];
    }];
    
}

@end

File: 结合转场动画及视频录制/结合转场动画及视频录制/Extensions/CustomAnimator.swift
//
//  CustomAnimator.swift
//  结合转场动画及视频录制
//
//  Created by 李森 on 2017/2/23.
//  Copyright © 2017年 李森. All rights reserved.
//

import UIKit

class CustomAnimator: NSObject, UIViewControllerAnimatedTransitioning {
    
    weak var transitionContext: UIViewControllerContextTransitioning?
    
    func transitionDuration(using transitionContext: UIViewControllerContextTransitioning?) -> TimeInterval {
        return 2.0
    }
    
    func animationDidStop(anim: CAAnimation!, finished flag: Bool) {
        self.transitionContext?.completeTransition(!(self.transitionContext?.transitionWasCancelled)!)
        
        self.transitionContext?.viewController(forKey: .from)?.view.layer.mask = nil
    }
    
    func animateTransition(using transitionContext: UIViewControllerContextTransitioning) {
        let containerView: UIView = transitionContext.containerView
        let toView: UIView = transitionContext.view(forKey: .to)!
        let fromView: UIView = transitionContext.view(forKey: .from)!
        toView.alpha = 0.0
        fromView.alpha = 1.0
        containerView.addSubview(toView)
        
        UIView.animate(withDuration: 2.0, animations: { 
            fromView.alpha = 0.0
            toView.alpha = 1.0
        }) { (isFinished: Bool) in
            transitionContext.completeTransition(true)
        }
    }
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
}
